test/auth.spec.js

```diff
@@ -0,0 +1,65 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const app = require('../server/index');
+const knex = require('../dbConn');
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+let register;
+
+describe('AUTH API ENDPOINTS', () => {
+  before(async () => {
+    await knex.migrate.latest();
+  });
+  beforeEach(() => {
+    register = {
+      firstName: 'john',
+      lastName: 'doe',
+      email: 'john@gmail.com',
+      password_confirmation: 'qwer1234',
+      password: 'qwer1234'
+    };
+  });
+  after(async () => {
+    await knex('users').truncate();
+    knex.destroy();
+    app.close();
+  });
+
+  describe('POST SIGN UP api/v1/auth/signup', () => {
+    it('should not sign up user if email is empty', async () => {
+      try {
```

Comment:
Don't wrap the tests in a try-catch. It's insecure. The aim of tests is to watch them fail if there are any errors.

Reply:
fixed. removed all try catch blocks. 
P.S: I implemented it in the first place because it solved the unhandled promise rejection error, and provided a neat way of emitting error/error stacks in the catch block.... would love to know another walk around for this problem.